func/internal/podevaluator.go

```diff
@@ -532,6 +650,7 @@ func (pm *podManager) retrieveOrCreatePod(ctx context.Context, image string, ttl
 	// TODO: It's possible to set up a Watch in the fn runner namespace, and always try to maintain a up-to-date local cache.
 	podList := &corev1.PodList{}
 	podTemplate, templateVersion, err := pm.getBasePodTemplate(ctx)
+	pm.appendImagePullSecret(image, registryAuthSecretPath, podTemplate)
```

Comment:
Should this happen after the err check below?

Reply:
amended in latest commit.